packages/components/src/Combobox/hooks/useCombobox.ts

```diff
@@ -38,22 +39,12 @@ export function useCombobox(
   const shouldScroll = useRef<boolean>(false);
   const [open, setOpen] = useState<boolean>(false);
   const [searchValue, setSearchValue] = useState<string>("");
-  const [internalFilteredOptions, setInternalFilteredOptions] =
-    useState<ComboboxOption[]>(options);
 
   const searchCallback = useCallback(
-    debounce((value: string) => onSearch && onSearch(value), debounceTime),
-    [],
+    debounce((value: string) => onSearch?.(value), debounceTime),
+    [onSearch, debounceTime],
   );
```

Comment:
ah whoops just saw this now

I think that's perfectly valid feedback. I am going to see about doing this in a slightly different way, but wanted to get something in ASAP because this is blocking a couple things at the moment

happy to take a look at this at the same time!

Reply:
that's a fair point. that is what we're trying to go for. wanna put a PR up for it?